sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -106,11 +108,20 @@ private[execution] object HashedRelation {
           1),
         0)
     }
-
-    if (key.length == 1 && key.head.dataType == LongType) {
-      LongHashedRelation(input, key, sizeEstimate, mm)
-    } else {
-      UnsafeHashedRelation(input, key, sizeEstimate, mm)
+    try {
+      if (key.length == 1 && key.head.dataType == LongType) {
+        LongHashedRelation(input, key, sizeEstimate, mm)
+      } else {
+        UnsafeHashedRelation(input, key, sizeEstimate, mm)
+      }
+    } catch {
+      case oe: SparkOutOfMemoryError =>
+        throw new SparkOutOfMemoryError(s"If this SparkOutOfMemoryError happens in Spark driver," +
```

Comment:
ah i see. So the `SparkOutOfMemoryError` is thrown by `BytesToBytesMap`, we need to catch and rethrow it to attach the error message anyway.

I also found that we may throw OOM when calling `child.executeCollectIterator` which calls `RDD#collect`, seems the previous code is corrected.

Reply:
So, I change back ?

Reply:
it seems we don't need to change anything, maybe just add some comments to say where OOM can occur, i.e. `RDD#collect` and `BroadcastMode#transform`